src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/entity/Alert.kt

```diff
@@ -105,18 +111,123 @@ data class Alert(
+    description: String?,
+    authorisedBy: String?,
+    activeFrom: LocalDate?,
+    activeTo: LocalDate?,
+    appendComment: String?,
+    updatedAt: LocalDateTime = LocalDateTime.now(),
+    updatedBy: String,
+    updatedByDisplayName: String,
+  ) = apply {
+    val descriptionUpdated = description != null && this.description != description
+    val authorisedByUpdated = authorisedBy != null && this.authorisedBy != authorisedBy
+    val activeFromUpdated = activeFrom != null && this.activeFrom != activeFrom
+    val activeToUpdated = this.activeTo != activeTo
+    val commentAppended = !appendComment.isNullOrEmpty()
+    var updated = false
+
+    val sb = StringBuilder()
+    if (descriptionUpdated) {
+      sb.appendLine("Updated alert description from '${this.description}' to '$description'")
+      this.description = description
+      updated = true
+    }
+    if (authorisedByUpdated) {
+      sb.appendLine("Updated authorised by from '${this.authorisedBy}' to '$authorisedBy'")
+      this.authorisedBy = authorisedBy
+      updated = true
+    }
+    if (activeFromUpdated) {
+      sb.appendLine("Updated active from from '${this.activeFrom}' to '$activeFrom'")
+      this.activeFrom = activeFrom!!
+      updated = true
+    }
+    if (activeToUpdated) {
+      sb.appendLine("Updated active to from '${this.activeTo}' to '$activeTo'")
+      this.activeTo = activeTo
+      updated = true
+    }
+    if (commentAppended) {
+      sb.appendLine("Comment '$appendComment' was added")
+      addComment(comment = appendComment!!, createdAt = updatedAt, createdBy = updatedBy, createdByDisplayName = updatedByDisplayName)
+      updated = true
+    }
+
+    if (updated) {
+      auditEvent(
+        action = AuditEventAction.UPDATED,
+        description = sb.toString(),
+        actionedAt = updatedAt,
+        actionedBy = updatedBy,
+        actionedByDisplayName = updatedByDisplayName,
+      )
+      registerEvent(
+        AlertUpdatedEvent(
+          alertUuid = alertUuid,
+          prisonNumber = prisonNumber,
+          alertCode = alertCode.code,
+          occurredAt = updatedAt,
+          source = NOMIS,
+          updatedBy = updatedBy,
+          descriptionUpdated = descriptionUpdated,
+          authorisedByUpdated = authorisedByUpdated,
+          activeFromUpdated = activeFromUpdated,
+          activeToUpdated = activeToUpdated,
+          commentAppended = commentAppended,
+        ),
+      )
+    }
+  }
+
   fun delete(
     deletedAt: LocalDateTime = LocalDateTime.now(),
     deletedBy: String,
     deletedByDisplayName: String,
   ): AuditEvent {
     this.deletedAt = deletedAt
-    return auditEvent(
+    val auditEvent = auditEvent(
       action = AuditEventAction.DELETED,
       description = "Alert deleted",
       actionedAt = deletedAt,
       actionedBy = deletedBy,
       actionedByDisplayName = deletedByDisplayName,
     )
+    registerEvent(
+      AlertDeletedEvent(
+        alertUuid = alertUuid,
```

Comment:
Minor, but arguably a better way to do this in Kotlin is something like:
```
return auditEvent(
 ...
).also {
  registerEvent(
    ...
  )
}
```
Makes it clearer that auditEvent is what we're interested in and registerEvent is a side effect. But like I said, very minor.

Reply:
I like it! Implemented